src/inc/Lfo.h

```diff
@@ -0,0 +1,88 @@
+#if !defined(__Lfo_hdr__)
+#define __Lfo_hdr__
+
+#define _USE_MATH_DEFINES
+#include <math.h>
+
+#include "ErrorDef.h"
+#include "Synthesis.h"
+#include "RingBuffer.h"
+
+class CLfo
+{
+public:
+    CLfo(float fModFreq, float fSampleRate)
+    {
+        m_pCRingBuffer = 0;
+        m_fRead = 0;
+        m_fModFreq = fModFreq;
+        m_fSampleRate = fSampleRate;
+        if (m_fModFreq == 0) {
+            m_iBufferLength = 1;
+        } else {
+            m_iBufferLength = static_cast<int>(1 / m_fModFreq);
```

Comment:
Is m_fModFreq a constant that will not change once the instance is created?

Reply:
No it will be updated when you call out the CLfo::setParam function.
But, besides that, I just found I lost a lot changes which hadn't been pushed.
In my newest version, it actually execute an reset() and init() evrery time when you want to use CVibrato::setParam function.

Reply:
The lost setParam/getParam function added, check the newest version.